core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1860,6 +1860,30 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   }
 
   /**
+   * Submit a job for execution and return a FutureJob holding the result. Return a Future for
+   * monitoring the job success or failure event.
+   */
+  private[spark] def submitAsyncJob[T, U, R](
```

Comment:
One high-level question, which perhaps has been addressed elsewhere: why can't we use the existing `submitJob` method that returns a `SimpleFutureAction`?